### ISSUE TITLE
Inability to write annotations on the outer class portion of an inner class name

### ISSUE THREAD

**Issue by jwaataja** [typetools/checker-framework#3562]
Dataflow seems to break with inner class types.
```java
import org.checkerframework.checker.tainting.qual.*;

public class C {
    void method(@Tainted Inner param) {
        @Tainted C. @Untainted Inner local = param;
    }

    class Inner {
    }
}
```
I get this error,
```
C.java:5: error: [type.invalid.conflicting.annos] invalid type: conflicting annotations [@Tainted, @Untainted] in type "@Tainted C.@Tainted @Untainted Inner"
        @Tainted C. @Untainted Inner local = param;
                                     ^
```
I would expect to instead get an invalid assignment error. It seems dataflow tries to replace the inner type but doesn't remove the old annotation.

**Comment by jwaataja** [comment]
I originally thought this had something to do with dataflow, but it didn't. Simplified test case:
```java
import org.checkerframework.checker.tainting.qual.*;

public class C {
    @Tainted C. @Untainted Inner field;

    class Inner {
    }
}
```
The [JLS](https://docs.oracle.com/javase/specs/jls/se14/html/jls-8.html#jls-8.3) specifies that on fields, the annotation immediately before the type is a modifier of the field itself, so it makes sense that both annotations are written on the `Inner` class. However, this raises the question of how to specify the type `@Tainted C. @Untainted Inner`. That is, an `@Untainted` instance of `Inner` where `C.this` has type `@Tainted`.

**Comment by smillst** [comment]
I've confirmed this is a bug in they way the Checker Framework decodes annotations from VariableTrees.  I'm working on fixing it.
